src/Diagram.php

```diff
@@ -144,20 +147,28 @@ public function create($entity_types) {
                 else {
                   $target_bundles = [];
                 }
-
-                if (!empty($target_bundles)) {
-                  $stop = null;
-                  foreach ($target_bundles as $target_bundle) {
-                    $entity_refs[$entity->id()][$bundle_name][$field_name][$reference_target_type] = [
-                      'bundle' => $target_bundle,
-                      'cardinality' => $instance_info->getFieldStorageDefinition()->getCardinality(),
-                      'required' => $instance_info->isRequired(),
-                      'fieldname' => $field_name,
-                    ];
-                  }
+              }
+              else {
+                // Get target info for other field types.
+                switch ($instance_info->getType()) {
```

Comment:
And in else clause we have a switch also on `$instance_info->getType()`

Reply:
updated